src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -75,35 +76,115 @@ internal sealed class HttpWebRequestDiagnosticSource : DiagnosticListener
         private static Func<HttpWebResponse, bool> isWebSocketResponseAccessor;
         private static Func<HttpWebResponse, string> connectionGroupNameAccessor;
 
-        // Fields for controlling initialization of the HttpWebRequestDiagnosticSource singleton
-        private bool initialized = false;
-
-        private HttpWebRequestDiagnosticSource()
-            : base(DiagnosticListenerName)
+        internal HttpWebRequestActivitySource()
         {
+            try
+            {
+                PrepareReflectionObjects();
+                PerformInjection();
+            }
+            catch (Exception ex)
+            {
+                // If anything went wrong, just no-op. Write an event so at least we can find out.
+                InstrumentationEventSource.Log.ExceptionInitializingInstrumentation(typeof(HttpWebRequestActivitySource).FullName, ex);
+            }
         }
 
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddRequestTagsAndInstrumentRequest(HttpWebRequest request, Activity activity)
         {
-            IDisposable result = base.Subscribe(observer, isEnabled);
-            this.Initialize();
-            return result;
+            activity.DisplayName = HttpTagHelper.GetOperationNameForHttpMethod(request.Method);
+
+            InstrumentRequest(request, activity);
+
+            activity.SetCustomProperty("HttpWebRequest.Request", request);
```

Comment:
Q.: why the property is set even if not recording? Similar for other methods.

Reply:
Short answer: No reason, just my gut feel for how it should be.

Long answer: The only reason to add these custom properties _at all_ is just in case someone downstream needs to add or change the data based on the real objects. If I don't add them in all cases (like it is currently doing), they won't be able to change anything when recording in the limited mode. I felt like we needed to add them always (if we're going to add them at all) so if downstream consumers wanted to have limited/full logic, they could. At the cost of an allocation to store the objects:

https://github.com/dotnet/runtime/blob/bd4697644e7024d1e5be94b674f73066bee112a4/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L816

Reply:
That is a good point. Perhaps we want to have some convention for the keys so child spans don't depend on a specific implementation handling the parent. I am not sure if this is necessary but we may want to provide some helper to "flatten" the search for the property: the child needs to navigate parents to search for this property.